Remove some StringMapWrapper facades

alexeagle opened the pull request:
Automated using https://github.com/alexeagle/tslint/commit/12012b07a2f7c341424aee5b40660ca0839bc347
with a few manual fixes after.

vicb commented on modules/@angular/benchpress/src/metric/multi_metric.ts:
```diff
@@ -57,8 +56,12 @@ export class MultiMetric extends Metric {
-  maps.forEach(
-      map => { StringMapWrapper.forEach(map, (value, prop) => { result[prop] = value; }); });
+  maps.forEach(map => {
+    Object.keys(map).forEach(prop => {
```
remove "{", convert to one liner

alexeagle replied:
The advantage of the curly braces is they either contain a return statement or are void. typescript-eng has discussed the style here and I think we've decided that arrow function with a bare expression on the RHS means you use the return value from it. Otherwise it is unclear if you are using an expression to mean void, or are using its return value.

vicb replied:
Ok for braces but still should be 1 liner,  wdyt

alexeagle replied:
It's up to clang-format to decide how many lines... do you just mean formatting, or do you mean to change syntax?

vicb replied:
Change the syntax, do not introduce the const

vicb commented on modules/@angular/benchpress/src/reporter/util.ts:
```diff
@@ -17,7 +17,10 @@ export function formatNum(n: number) {
 export function sortedProps(obj: {[key: string]: any}) {
   var props: string[] = [];
-  StringMapWrapper.forEach(obj, (value, prop) => props.push(prop));
+  Object.keys(obj).forEach(prop => {
```
unused value, one liner

alexeagle replied:
I fixed these cases where I created a value variable that was not used, PTAL

vicb replied:
Push(...Object.keys)

vicb commented on modules/@angular/benchpress/src/sample_description.ts:
```diff
@@ -42,7 +41,10 @@ export class SampleDescription {
     this.description = {};
     descriptions.forEach(description => {
-      StringMapWrapper.forEach(description, (value, prop) => this.description[prop] = value);
+      Object.keys(description).forEach(prop => {
```
again

vicb replied:
1 line

vicb commented on modules/@angular/benchpress/test/metric/perflog_metric_spec.ts:
```diff
@@ -68,7 +67,10 @@ export function main() {
     function sortedKeys(stringMap: {[key: string]: any}) {
       var res: string[] = [];
-      StringMapWrapper.forEach(stringMap, (_, key) => { res.push(key); });
+      Object.keys(stringMap).forEach(key => {
```
...

vicb replied:
Unused

vicb replied:
Res.push(...Object.keys)

vicb commented on modules/@angular/compiler/src/animation/animation_parser.ts:
```diff
@@ -338,13 +338,12 @@ function _parseAnimationKeyframes(
-            }
-          });
+      Object.keys(entry).forEach(prop => {
+        const value: any = (entry as{[key: string]: string | number})[prop];
```
Move in if

vicb commented on modules/@angular/compiler/src/animation/animation_parser.ts:
```diff
@@ -381,24 +380,23 @@ function _parseAnimationKeyframes(
-          }
-        });
+    Object.keys(styles).forEach(prop => {
+      const value = styles[prop];
```
Move in if

vicb reviewed (commented):
Most 2 liners should be converted to 1 line IMO
Did not repeat the same comments towards the end

alexeagle committed:
refactor(facade): Remove most of StringMapWrapper facade.

This change mostly automated by
https://github.com/alexeagle/tslint/commit/12012b07a2f7c341424aee5b40660ca0839bc347
with some manual fixes.

alexeagle commented:
I did a bunch more manual cleanup, mostly removing the introduced `const value` if it was only used once.

vicb commented:
@alexeagle could you please rebase

alexeagle commented:
rebased

On Mon, Oct 3, 2016 at 3:45 PM Victor Berchet <notifications@github.com>
wrote:

> @alexeagle <https://github.com/alexeagle> could you please rebase
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/pull/12022#issuecomment-251248927>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAC5I1T7dOEv854HPO_TozfI5-Do6ET8ks5qwYV9gaJpZM4KLn6b>
> .
>
